cmd/compute-domain-kubelet-plugin/device_state.go

```diff
@@ -153,42 +156,57 @@ func (s *DeviceState) Prepare(ctx context.Context, claim *resourceapi.ResourceCl
 		return nil, fmt.Errorf("unable to create CDI spec file for claim: %w", err)
 	}
 
-	preparedClaims[claimUID] = preparedDevices
+	// Add ResourceClaim API object to checkpoint so that for Unprepare() only
```

Comment:
Are there situations when we should also query the API server (e.g. if the local state is not sufficient?).

Reply:
I wonder the same. That's certainly a key question here.

There _should not_ be such situations. Conceptually, we have to build proper cleanup w/o API server input (input other than ResourceClaim UID). I have edited the PR description a bit.